File: LeetCode/0104. Maximum Depth of Binary Tree/Solution.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int = 0) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

/**
 * 38 / 38 test case passed
 * Status: Accepted
 * Runtime: 394 ms
 */
class Solution {
    fun maxDepth(root: TreeNode?): Int {
        if (null == root) return 0
        return maxOf(maxDepth(root.left), maxDepth(root.right)) + 1
    }
}

File: LeetCode/0104. Maximum Depth of Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

"""
39 / 39 test case passed
Status: Accepted
Runtime: 56 ms
"""
class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        return 0 if root is None else max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1